src/mqtt.cpp

```diff
@@ -255,12 +260,55 @@ namespace mqtt {
     }
   }
 
+  // Queue a status message with provided document.
+  // This takes ownership (and will free) the provided document after serialization.
+  // If the document does not contain a key name 'online', it will be added and set
+  // to true.
+  // The document is serialized into a buffer of STATUS_BUFSIZE bytes, so you should
+  // make sure to keep your document below that.
+  void sendStatus(DynamicJsonDocument *status) {
+    MqttMessage msg;
+    msg.cmd = X_CMD_PUBLISH_STATUS;
+    msg.status = status;
+    if (mqttQueue) xQueueSendToBack(mqttQueue, (void*)&msg, pdMS_TO_TICKS(100));
+  }
+
+  // Helper for sending a simple status message. Must be shorter than
+  //  STATUS_BUFSZIZE - 48 (e.g. 464 chars by default) to allow for JSON overhead.
+  void sendStatusMsg(const char *msg) {
+    DynamicJsonDocument *status = new DynamicJsonDocument(STATUS_BUFSIZE);
+    (*status)["status"] = msg;
+    if (status->overflowed()) {
+      ESP_LOGW(TAG, "Status overflowed buffer, may not be sent: %s", msg);
+    }
+    sendStatus(status);
+  }
+
+  void publishStatusInternal(DynamicJsonDocument *status) {
+    if (!mqtt_client->connected()) {
+      return;
```

Comment:
This is missing `delete status;` => memory leak

Reply:
good catch x2